File: Iterator/MenuItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Collections;

namespace Iterator
{
    public class MenuItem : IMenu
    {
        string name;
        public string Name
        {
            get { return name; }
            set { name = value; }
        }

        string description;
        public string Description
        {
            get { return description; }
            set { description = value; }
        }

        bool isVegetarian;
        public bool IsVegetarian
        {
            get { return isVegetarian; }
            set { isVegetarian = value; }
        }

        double price;
        public double Price
        {
            get { return price; }
            set { price = value; }
        }

        public MenuItem(string name, string description, bool isVegetarian, double price)
        {
            this.Name = name;
            this.Description = description;
            this.IsVegetarian = isVegetarian;
            this.Price = price;
        }

        public void Print()
        {
            System.Console.WriteLine(
                "Menu Item\n-------------\nVegetarian? {0}\n{1}, {2} -- {3}",
                this.IsVegetarian, this.Name, this.Price, this.Description);
        }
    }
}

File: Composite/src/com/jkhines/DesignPatterns/CompositePattern/MenuItem.java
package com.jkhines.DesignPatterns.CompositePattern;

public class MenuItem extends MenuNode {
	private String name;
	private String description;
	private boolean isVegetarian;
	private double price;
	
	public MenuItem(String name, String description, boolean isVegetarian, double price) {
		this.name = name;
		this.description = description;
		this.isVegetarian = isVegetarian;
		this.price = price;
	}

	@Override
	public String getName() {
		return name;
	}

	@Override
	public String getDescription() {
		return description;
	}

	@Override
	public boolean isVegetarian() {
		return isVegetarian;
	}

	@Override
	public double getPrice() {
		return price;
	}
	
	@Override
	public void print() {
		System.out.println(getName());
		System.out.println(getDescription());
		System.out.println(isVegetarian());
		System.out.println(getPrice() + "\n");
	}
}
